### ISSUE TITLE
Modification of yolov8 

### ISSUE THREAD

**Issue by Abdullah21533** [ultralytics/ultralytics#16740]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Hyperparameter Tuning

### Bug

I am making changes to the YOLOv8 model's YAML file, but I am not seeing any improvements reflected. What steps should I take to ensure that the improvements are correctly applied and visible?

### Environment

Ultralytics YOLOv8.0.196 🚀 Python-3.10.12 torch-2.4.1+cu121 CUDA:0 (NVIDIA L4, 22700MiB)
Setup complete ✅ (12 CPUs, 53.0 GB RAM, 36.4/112.6 GB disk)

### Minimal Reproducible Example

```
# Custom YOLOv8 with DSConv, DPAG, and FEM

# Parameters
nc: 80 # number of classes
scales:
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024]
  s: [0.33, 0.50, 1024]
  m: [0.67, 0.75, 768]
  l: [1.00, 1.00, 512]
  x: [1.00, 1.25, 512]

# Custom backbone with DSConv and FEM
backbone:
  # [from, repeats, module, args]
  - [-1, 1, DSConv, [64, 3, 2]] # 0-P1/2 (Replaced Conv with DSConv)
  - [-1, 1, DSConv, [128, 3, 2]] # 1-P2/4
  - [-1, 3, FEM, [128, True]]    # Feature Enhancement Module (FEM)
  - [-1, 1, DSConv, [256, 3, 2]] # 3-P3/8
  - [-1, 6, C2f, [256, True]]    # Standard C2f block
  - [-1, 1, DSConv, [512, 3, 2]] # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, DSConv, [1024, 3, 2]] # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]] # 9

# Custom head with DPAG
head:
  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 6], 1, Concat, [1]] # cat backbone P4
  - [-1, 3, DPAG, [512]]      # DPAG replaces C2f block for better attention

  - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
  - [[-1, 4], 1, Concat, [1]] # cat backbone P3
  - [-1, 3, DPAG, [256]]      # DPAG for attention (P3/8-small)

  - [-1, 1, DSConv, [256, 3, 2]]  # Replace Conv with DSConv
  - [[-1, 12], 1, Concat, [1]] # cat head P4
  - [-1, 3, DPAG, [512]]      # DPAG for better attention (P4/16-medium)

  - [-1, 1, DSConv, [512, 3, 2]]  # Replace Conv with DSConv
  - [[-1, 9], 1, Concat, [1]] # cat head P5
  - [-1, 3, DPAG, [1024]]     # DPAG for better attention (P5/32-large)

  - [[15, 18, 21], 1, Detect, [nc]] # Detect(P3, P4, P5)
```

### Additional

I am making changes to the YOLOv8 model's YAML file, but I am not seeing any improvements reflected. What steps should I take to ensure that the improvements are correctly applied and visible?

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Abdullah21533, thank you for your interest in Ultralytics 🚀! It seems you're diving into some exciting model customization. For new users, we recommend checking out our [Docs](https://docs.ultralytics.com) for a wealth of resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

As this appears to be a 🐛 Bug Report, please make sure your provided [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) captures all relevant details. This will help us assist you more effectively.

If you have any custom training ❓ questions, more details like dataset examples and training logs are always helpful. Ensure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join us in the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com)
- Or share knowledge on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Make sure you're using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment. Use the following command to upgrade:

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in various environments with all dependencies set up. Here are a few options:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests ensure correct operation across platforms and tasks.

This is an automated response. An Ultralytics engineer will be here to assist you soon.

**Comment by Skillnoob** [comment]
In general I would not recommend modifying the yaml's in any way unless you know what you are doing. If you are aiming for better accuracy I'd rather advise you to improve your dataset or simply using a larger model size. If you do want to use a different yaml than the default yolov8 model, there is a wide variety of modified yolov8 yaml's available [here](https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/models/v8). I've seen the `-ghost` models perform well.

**Comment by Mahfuzkiron** [comment]
> In general I would not recommend modifying the yaml's in any way unless you know what you are doing. If you are aiming for better accuracy I'd rather advise you to improve your dataset or simply using a larger model size. If you do want to use a different yaml than the default yolov8 model, there is a wide variety of modified yolov8 yaml's available [here](https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/models/v8). I've seen the `-ghost` models perform well.

@Skillnoob hello bro, can i get your email address? i need some help about this problem

**Comment by Mahfuzkiron** [comment]
> ### Search before asking
> * [x]  I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.
> 
> ### Ultralytics YOLO Component
> Hyperparameter Tuning
> 
> ### Bug
> I am making changes to the YOLOv8 model's YAML file, but I am not seeing any improvements reflected. What steps should I take to ensure that the improvements are correctly applied and visible?
> 
> ### Environment
> Ultralytics YOLOv8.0.196 🚀 Python-3.10.12 torch-2.4.1+cu121 CUDA:0 (NVIDIA L4, 22700MiB) Setup complete ✅ (12 CPUs, 53.0 GB RAM, 36.4/112.6 GB disk)
> 
> ### Minimal Reproducible Example
> ```
> # Custom YOLOv8 with DSConv, DPAG, and FEM
> 
> # Parameters
> nc: 80 # number of classes
> scales:
>   # [depth, width, max_channels]
>   n: [0.33, 0.25, 1024]
>   s: [0.33, 0.50, 1024]
>   m: [0.67, 0.75, 768]
>   l: [1.00, 1.00, 512]
>   x: [1.00, 1.25, 512]
> 
> # Custom backbone with DSConv and FEM
> backbone:
>   # [from, repeats, module, args]
>   - [-1, 1, DSConv, [64, 3, 2]] # 0-P1/2 (Replaced Conv with DSConv)
>   - [-1, 1, DSConv, [128, 3, 2]] # 1-P2/4
>   - [-1, 3, FEM, [128, True]]    # Feature Enhancement Module (FEM)
>   - [-1, 1, DSConv, [256, 3, 2]] # 3-P3/8
>   - [-1, 6, C2f, [256, True]]    # Standard C2f block
>   - [-1, 1, DSConv, [512, 3, 2]] # 5-P4/16
>   - [-1, 6, C2f, [512, True]]
>   - [-1, 1, DSConv, [1024, 3, 2]] # 7-P5/32
>   - [-1, 3, C2f, [1024, True]]
>   - [-1, 1, SPPF, [1024, 5]] # 9
> 
> # Custom head with DPAG
> head:
>   - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
>   - [[-1, 6], 1, Concat, [1]] # cat backbone P4
>   - [-1, 3, DPAG, [512]]      # DPAG replaces C2f block for better attention
> 
>   - [-1, 1, nn.Upsample, [None, 2, "nearest"]]
>   - [[-1, 4], 1, Concat, [1]] # cat backbone P3
>   - [-1, 3, DPAG, [256]]      # DPAG for attention (P3/8-small)
> 
>   - [-1, 1, DSConv, [256, 3, 2]]  # Replace Conv with DSConv
>   - [[-1, 12], 1, Concat, [1]] # cat head P4
>   - [-1, 3, DPAG, [512]]      # DPAG for better attention (P4/16-medium)
> 
>   - [-1, 1, DSConv, [512, 3, 2]]  # Replace Conv with DSConv
>   - [[-1, 9], 1, Concat, [1]] # cat head P5
>   - [-1, 3, DPAG, [1024]]     # DPAG for better attention (P5/32-large)
> 
>   - [[15, 18, 21], 1, Detect, [nc]] # Detect(P3, P4, P5)
> ```
> 
> ### Additional
> I am making changes to the YOLOv8 model's YAML file, but I am not seeing any improvements reflected. What steps should I take to ensure that the improvements are correctly applied and visible?
> 
> ### Are you willing to submit a PR?
> * [x]  Yes I'd like to help by submitting a PR!

@Abdullah21533 hello bro, can i get your email address? i need some help about this problem
